src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts

```diff
@@ -45,8 +49,29 @@ export class DokobitAuthenticationProvider implements AuthenticationProvider {
 
     return {
       code: sessionStatus.code,
-      firstName: sessionStatus.name,
-      lastName: sessionStatus.surname,
+      profile: mapToProfile(sessionStatus),
     };
   }
 }
+
+function mapToProfile(sessionStatus: DokobitSessionStatus): Profile {
+  var id = getAndValidateEstonianIdCode(sessionStatus);
+  id.validate();
+  return new Profile(sessionStatus.name, sessionStatus.surname, DateOfBirth.fromDate(id.getBirthday()), getSex(id));
+}
+
+function getAndValidateEstonianIdCode(sessionStatus: DokobitSessionStatus): Isikukood {
+  const idCode = new Isikukood(sessionStatus.code);
+  if (!idCode.validate()) {
+    throw new AuthenticationError('idCode is not valid');
+  }
+  return idCode;
+}
+
+function getSex(id: Isikukood): Sex {
+  const gender = id.getGender();
+  if (gender === 'unknown') {
+    throw new AuthenticationError('Non parsable gender');
+  }
+  return gender === 'male' ? Sex.MALE : Sex.FEMALE;
```

Comment:
Perhaps we could be more explicit with determining female, e.g.:
```suggestion
  const gender = id.getGender();
  
  switch (gender) {
    case 'male':
      return Sex.MALE;
    case 'female':
      return Sex.FEMALE;
    default:
      throw new AuthenticationError('Non parsable gender');    
  }
```

Reply:
yeah, much cleaner